### DISCUSSION TITLE
How to change the decision on default language for file extension?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#194] [General]
I accidentally chose Visual Basic for `.bas` file extension and it will treat `.bas` files as Visual Basic without asking me again.

**Comment by SpartanJ** [comment]
You can't 🥲. I should add an option somewhere, this needs design. Let me think about it.
Meanwhile you can edit the config file manually.
Open the ecode config file: config.cfg, that it's located at the configuration folder path:
Linux: uses XDG_CONFIG_HOME, usually translates to ~/.config/ecode/
macOS: uses Application Support folder in HOME, usually translates to ~/Library/Application Support/ecode/
Windows: uses APPDATA, usually translates to C:\Users\{username}\AppData\Roaming\ecode\

Search for: `[languages_extensions]` and remove the block or edit it (it will have something like `bas=visualbasic`. Also edit it with other editor than ecode, because on ecode close it will override your changes :thinking: .

**Reply by ghost** [reply]
> I should add an option somewhere, this needs design.

You don't need to. I'm OK with the current behavior. Still better than asking me each time I open a `.bas` file like on other editors. I'm not afraid of editing config files, either.
